src/ValidationError.js

```diff
@@ -1,23 +1,854 @@
+const SPECIFICITY = {
+  type: 1,
+  not: 1,
+  oneOf: 1,
+  anyOf: 1,
+  if: 1,
+  enum: 1,
+  const: 1,
+  instanceof: 1,
+  required: 2,
+  pattern: 2,
+  patternRequired: 2,
+  format: 2,
+  formatMinimum: 2,
+  formatMaximum: 2,
+  minimum: 2,
+  exclusiveMinimum: 2,
+  maximum: 2,
+  exclusiveMaximum: 2,
+  multipleOf: 2,
+  uniqueItems: 2,
+  contains: 2,
+  minLength: 2,
+  maxLength: 2,
+  minItems: 2,
+  maxItems: 2,
+  minProperties: 2,
+  maxProperties: 2,
+  dependencies: 2,
+  propertyNames: 2,
+  additionalItems: 2,
+  additionalProperties: 2,
+  absolutePath: 2,
+};
+
+function filterMax(array, fn) {
+  const evaluatedMax = array.reduce((max, item) => Math.max(max, fn(item)), 0);
+
+  return array.filter((item) => fn(item) === evaluatedMax);
+}
+
+function filterChildren(children) {
+  let newChildren = children;
+
+  newChildren = filterMax(newChildren, (error) =>
+    error.dataPath ? error.dataPath.length : 0
+  );
+  newChildren = filterMax(
+    newChildren,
+    (error) => SPECIFICITY[error.keyword] || 2
+  );
+
+  return newChildren;
+}
+
+function indent(str, prefix) {
+  return str.replace(/\n(?!$)/g, `\n${prefix}`);
+}
+
+function isObject(maybyObj) {
+  return typeof maybyObj === 'object' && maybyObj !== null;
```

Comment:
```diff
- return typeof maybyObj === 'object' && maybyObj !== null;
+ return typeof maybyObj === 'object';
```

Do we need `maybeObj !== null`?

Reply:
`typeof` null is object, so we should avoid this behavior